tinker/e_announcements/__init__.py

```diff
@@ -318,4 +340,21 @@ def get_title_and_message(form):
 
     # TODO e-announcements by role (someday)
 
+    @route("/search", methods=['POST'])
+    def search(self):
+        data = self.base.dictionary_encoder.encode(json.loads(request.data))
+        selection = data['selection']
+        title = data['title']
+        date = data['date']
+        try:
+            date = datetime.datetime.strptime(date, "%a %b %d %Y")
+
+        except:
+            # Set start and end to be false so that hasDates is set to false
+            date = 0
+
+        search_results, forms_header = self.base.get_search_results(selection, title, date)
+        search_results.sort(key=lambda item: datetime.datetime.strptime(item['first_date'], '%A %B %d, %Y'), reverse=True)
+        return render_template('e-announcements/results.html', list_of_annz=search_results, formsHeader=forms_header)
```

Comment:
you could probably use **locals() here. We typically just use that, so it would keep it consistent.

Reply:
Done